hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -475,27 +465,23 @@ findParticipationTokens headCurrency (Value val) =
 
 mustContinueHeadWith :: ScriptContext -> Address -> Integer -> Datum -> Bool
 mustContinueHeadWith ScriptContext{scriptContextTxInfo = txInfo} headAddress changeValue datum =
-  checkOutputDatumAndValue [] (txInfoOutputs txInfo)
+  case txInfoOutputs txInfo of
+    -- NOTE: in the real scenario here we should always get two outputs - head
+    -- and change one. But, since we are not dealing with the change outputs in
+    -- tests we either need to keep this pattern match for a single head output
+    -- or fix the end-to-end spec to actually add a change output and have more
+    -- realistic txs.
+    [headOutput] ->
+      checkHeadOutput headOutput
+    [headOutput, changeOutput] ->
```

Comment:
I never understood why this function would check the `changeValue` and consequently another function would need to aggregate that `changeValue` in the first place.

This now is also leading us to add artificial change outputs to make a transaction valid (see comment above). Hence, I would argue for:

Must: not reason about the `changeOutput` nor the `changeValue` of a transaction.

I would even say the proper way forward here is to change the `checkCollectCom` to align better with the statements from the spec to use `checkHeadOutput`, but also extract the `eta'` from the datum and compare it with the `utxoHash` from line 203.